streams/src/main/java/org/apache/kafka/streams/processor/internals/Tasks.java

```diff
@@ -52,11 +53,8 @@ class Tasks implements TasksRegistry {
     // we receive a new assignment and they are revoked from the thread.
     private final Map<TaskId, Set<TopicPartition>> pendingActiveTasksToCreate = new HashMap<>();
     private final Map<TaskId, Set<TopicPartition>> pendingStandbyTasksToCreate = new HashMap<>();
-    private final Map<TaskId, Set<TopicPartition>> pendingTasksToRecycle = new HashMap<>();
-    private final Map<TaskId, Set<TopicPartition>> pendingTasksToUpdateInputPartitions = new HashMap<>();
     private final Set<Task> pendingTasksToInit = new HashSet<>();
-    private final Set<TaskId> pendingTasksToCloseClean = new HashSet<>();
-    private final Set<TaskId> pendingTasksToCloseDirty = new HashSet<>();
+    private final Map<TaskId, PendingUpdateAction> pendingUpdateActions = new HashMap<>();
```

Comment:
This map ensures that each task has at most one pending update action. The last assigned pending update action is stored in the map.